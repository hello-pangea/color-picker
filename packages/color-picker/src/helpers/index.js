export * from "./checkboard";
export * from "./color";
