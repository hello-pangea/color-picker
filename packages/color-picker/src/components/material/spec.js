/* global test, test, expect */

import React from "react";
import renderer from "react-test-renderer";
import { red } from "../../helpers/color";

import Material from "./Material";

test("Material renders correctly", () => {
  const tree = renderer.create(<Material {...red} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test("Material renders custom styles correctly", () => {
  const tree = renderer
    .create(
      <Material
        {...red}
        styles={{ default: { wrap: { boxShadow: "0 0 10px red" } } }}
      />
    )
    .toJSON();
  expect(tree.props.style.boxShadow).toBe("0 0 10px red");
});
