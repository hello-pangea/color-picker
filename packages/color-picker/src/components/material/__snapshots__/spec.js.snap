// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Material renders correctly 1`] = `
<div
  style={
    {
      "display": "inline-block",
      "position": "relative",
    }
  }
>
  <div
    style={
      {
        "background": "#fff",
        "borderRadius": 2,
        "boxShadow": "0 2px 10px rgba(0,0,0,.12), 0 2px 5px rgba(0,0,0,.16)",
        "inset": "0px",
        "position": "absolute",
      }
    }
  />
  <div
    style={
      {
        "position": "relative",
      }
    }
  >
    <div
      className="material-picker "
      style={
        {
          "fontFamily": "Roboto",
          "height": "98px",
          "padding": "16px",
          "width": "98px",
        }
      }
    >
      <div
        style={
          {
            "position": "relative",
          }
        }
      >
        <input
          id="rc-editable-input-1"
          onBlur={[Function]}
          onChange={[Function]}
          onKeyDown={[Function]}
          spellCheck="false"
          style={
            {
              "border": "0px",
              "borderBottom": "2px solid #22194d",
              "color": "#333",
              "fontSize": "15px",
              "height": "30px",
              "marginTop": "12px",
              "outline": "none",
              "padding": "0px",
              "width": "100%",
            }
          }
          value="#22194D"
        />
        <label
          htmlFor="rc-editable-input-1"
          onMouseDown={[Function]}
          style={
            {
              "color": "#999999",
              "cursor": undefined,
              "fontSize": "11px",
              "left": "0px",
              "position": "absolute",
              "textTransform": "capitalize",
              "top": "0px",
            }
          }
        >
          hex
        </label>
      </div>
      <div
        className="flexbox-fix"
        style={
          {
            "display": "flex",
            "marginRight": "-10px",
            "paddingTop": "11px",
          }
        }
      >
        <div
          style={
            {
              "flex": "1",
              "paddingRight": "10px",
            }
          }
        >
          <div
            style={
              {
                "position": "relative",
              }
            }
          >
            <input
              id="rc-editable-input-2"
              onBlur={[Function]}
              onChange={[Function]}
              onKeyDown={[Function]}
              spellCheck="false"
              style={
                {
                  "border": "0px",
                  "borderBottom": "1px solid #eee",
                  "color": "#333",
                  "fontSize": "15px",
                  "height": "30px",
                  "marginTop": "12px",
                  "outline": "none",
                  "padding": "0px",
                  "width": "100%",
                }
              }
              value="34"
            />
            <label
              htmlFor="rc-editable-input-2"
              onMouseDown={[Function]}
              style={
                {
                  "color": "#999999",
                  "cursor": undefined,
                  "fontSize": "11px",
                  "left": "0px",
                  "position": "absolute",
                  "textTransform": "capitalize",
                  "top": "0px",
                }
              }
            >
              r
            </label>
          </div>
        </div>
        <div
          style={
            {
              "flex": "1",
              "paddingRight": "10px",
            }
          }
        >
          <div
            style={
              {
                "position": "relative",
              }
            }
          >
            <input
              id="rc-editable-input-3"
              onBlur={[Function]}
              onChange={[Function]}
              onKeyDown={[Function]}
              spellCheck="false"
              style={
                {
                  "border": "0px",
                  "borderBottom": "1px solid #eee",
                  "color": "#333",
                  "fontSize": "15px",
                  "height": "30px",
                  "marginTop": "12px",
                  "outline": "none",
                  "padding": "0px",
                  "width": "100%",
                }
              }
              value="25"
            />
            <label
              htmlFor="rc-editable-input-3"
              onMouseDown={[Function]}
              style={
                {
                  "color": "#999999",
                  "cursor": undefined,
                  "fontSize": "11px",
                  "left": "0px",
                  "position": "absolute",
                  "textTransform": "capitalize",
                  "top": "0px",
                }
              }
            >
              g
            </label>
          </div>
        </div>
        <div
          style={
            {
              "flex": "1",
              "paddingRight": "10px",
            }
          }
        >
          <div
            style={
              {
                "position": "relative",
              }
            }
          >
            <input
              id="rc-editable-input-4"
              onBlur={[Function]}
              onChange={[Function]}
              onKeyDown={[Function]}
              spellCheck="false"
              style={
                {
                  "border": "0px",
                  "borderBottom": "1px solid #eee",
                  "color": "#333",
                  "fontSize": "15px",
                  "height": "30px",
                  "marginTop": "12px",
                  "outline": "none",
                  "padding": "0px",
                  "width": "100%",
                }
              }
              value="77"
            />
            <label
              htmlFor="rc-editable-input-4"
              onMouseDown={[Function]}
              style={
                {
                  "color": "#999999",
                  "cursor": undefined,
                  "fontSize": "11px",
                  "left": "0px",
                  "position": "absolute",
                  "textTransform": "capitalize",
                  "top": "0px",
                }
              }
            >
              b
            </label>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
